### PR TITLE
Refactoring and final javadoc additions

### PR DESCRIPTION
**Pull request by Bensonwu02**
This pull request contains
- final javadoc additions
- updated README
- case handled for non-numerical data input for goal setting and add metric
- addition of 'example metrics' folder for more example data.

### CODE REVIEW

File: src/main/java/models/SetGoalRequestModel.java [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+    /**
+     * Getter for goal
+     * @return  goal stored by the request model
+     */
     public double getGoal(){return this.goal;}
 
+    /**
+     * Getter for metric name
+     * @return  metric name string
+     */
```
Comment (craftedswift) [inline src/main/java/models/SetGoalRequestModel.java:30]: for java doc make sure there is a space between the method description and the @param/@return

File: src/main/java/presenters/MetricSumViewModel.java [hunk 2]
```diff
@@ -41,14 +41,30 @@ public XYChart getChart() {
     public String getMetricStatistics() {
         return metricStatistics;
     }
+
```
Comment (craftedswift) [inline src/main/java/presenters/MetricSumViewModel.java:44]: for java doc make sure there is a space between the method description and the @param/@return

File: src/main/java/use_cases/GoalSetter.java [hunk 3]
```diff
@@ -15,6 +17,11 @@ public GoalSetter(MetricStorageInterface metricStorage, SetGoalOutputBoundary pr
         this.presenter = presenter;
     }
 
+    /**
+     * Method for setting goal.
```
Comment (craftedswift) [inline src/main/java/use_cases/GoalSetter.java:21]: for java doc make sure there is a space between the method description and the @param/@return

**Review by tylerkj1209 — DISMISSED** [review]
Thank you for altering the fail view message for dataLogger to more clearly match with the actual exception being raised by the code. I see no issues with this pull request. Lots of good reworking of the documentation.

File: src/main/java/screens/ChooseMetricSumScreen.java [hunk 4]
```diff
@@ -74,6 +74,12 @@ public void actionPerformed(ActionEvent evt) {
         }
     }
 
+    /**
+     * A helper method to extract data contained in the view model which is used to make the MetricSummaryScreen.
+     * @param viewModel represents view model from which this method extracts information to construct
```
Comment (craftedswift) [inline src/main/java/screens/ChooseMetricSumScreen.java:79]: for java doc make sure there is a space between the method description and the @param/@return

File: src/main/java/use_cases/MetricDelInputBoundary.java [hunk 5]
```diff
@@ -3,6 +3,15 @@
 import models.MetricDelRequestModel;
 import models.MetricDelResponseModel;
 
+/**
+ * Interface for the Delete Metric use case
+ */
 public interface MetricDelInputBoundary {
+
+    /**
+     *
```
Comment (craftedswift) [inline src/main/java/use_cases/MetricDelInputBoundary.java:12]: Missing method description (my bad)

File: src/main/java/models/SetGoalRequestModel.java (hunk 1 above)
Comment (Bensonwu02) [reply src/main/java/models/SetGoalRequestModel.java:30]: resolved

File: src/main/java/use_cases/GoalSetter.java (hunk 3 above)
Comment (Bensonwu02) [reply src/main/java/use_cases/GoalSetter.java:21]: resolved

File: src/main/java/screens/ChooseMetricSumScreen.java (hunk 4 above)
Comment (Bensonwu02) [reply src/main/java/screens/ChooseMetricSumScreen.java:79]: resolved

File: src/main/java/use_cases/MetricDelInputBoundary.java (hunk 5 above)
Comment (Bensonwu02) [reply src/main/java/use_cases/MetricDelInputBoundary.java:12]: resolved

File: src/main/java/use_cases/SetGoalInputBoundary.java [hunk 6]
```diff
@@ -3,8 +3,16 @@
 
 import models.SetGoalRequestModel;
 
+/**
+ * Interface for the Set Goal use case
+ */
 public interface SetGoalInputBoundary {
 
+    /**
+     *
```
[the remark this answers is not in the document]
Comment (Bensonwu02) [reply src/main/java/use_cases/SetGoalInputBoundary.java:12]: resolved

**Comment by Bensonwu02** [comment]
Made changes in accordance with sanford's comments

File: README.md [hunk 7]
```diff
@@ -1,45 +1,39 @@
 # Project Mewdy
 
-Mewdy is a configurable mood tracking application with customizable metric tracker.
+Mewdy is a configurable metric tracking application.
 
 ## How to run:
 - run Main.java
```
Comment (Wizard603) [inline README.md:6]: Maybe mention there is a downloadable jar file in the releases tab

File: src/main/java/controllers/DataExportController.java [hunk 8]
```diff
@@ -8,15 +8,15 @@
  * Controller class for DataExporter
  */
 public class DataExportController {
-    private final DataExportInputBoundary INPUTBOUNDARY;
+    private final DataExportInputBoundary inputBoundary;
```
Comment (Wizard603) [inline src/main/java/controllers/DataExportController.java:11]: this is final so it should be all caps

File: src/main/java/controllers/DataImportController.java [hunk 9]
```diff
@@ -8,15 +8,15 @@
  * Controller for the DataImport use case
  */
 public class DataImportController {
-    private final DataImportInputBoundary INPUTBOUNDARY;
+    private final DataImportInputBoundary inputBoundary;
```
Comment (Wizard603) [inline src/main/java/controllers/DataImportController.java:11]: capitalize final variables

File: src/main/java/controllers/EntryUndoController.java [hunk 10]
```diff
@@ -9,7 +9,7 @@
  */
 public class EntryUndoController {
 
-    private EntryUndoInputBoundary entryUndo;
+    private final EntryUndoInputBoundary entryUndo;
```
Comment (Wizard603) [inline src/main/java/controllers/EntryUndoController.java:12]: capitalize final variables

File: src/main/java/entities/DataPoint.java [hunk 11]
```diff
@@ -8,9 +8,9 @@
  */
 public class DataPoint {
 
-    private final DateTimeFormatter FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-    private final double VALUE;
-    private final LocalDateTime DATE;
+    private final DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
```
Comment (Wizard603) [inline src/main/java/entities/DataPoint.java:11]: capitalize final variables

**Review by Wizard603 — CHANGES_REQUESTED** [review]
capitalize final variables

File: src/main/java/controllers/DataExportController.java (hunk 8 above)
Comment (Bensonwu02) [reply src/main/java/controllers/DataExportController.java:11]: only static variables are caps

File: src/main/java/controllers/DataImportController.java (hunk 9 above)
Comment (Bensonwu02) [reply src/main/java/controllers/DataImportController.java:11]: same as above

File: src/main/java/entities/DataPoint.java (hunk 11 above)
Comment (Bensonwu02) [reply src/main/java/entities/DataPoint.java:11]: same as above

File: README.md (hunk 7 above)
Comment (Bensonwu02) [reply README.md]: done

File: src/main/java/use_cases/DataImporter.java [hunk 12]
```diff
@@ -89,10 +89,10 @@ private ImportResponseModel read() {
                 createMetric(dates, data, upperBound, lowerBound, hasGoal, goal, filename);
                 storage.save(); // State should be identical from last save
             }
-        } catch (RuntimeException | IOException e) {
+        } catch (IOException e) {
             return presenter.prepareFailView("Folder either does not contain csv files or does not have access.");
-        } catch (ParseException e) {
-            return presenter.prepareFailView("Bad data.");
+        } catch (RuntimeException | ParseException e) {
```
Comment (Wizard603) [inline src/main/java/use_cases/DataImporter.java:94]: Thank you for making the message more clear

**Review by Wizard603 — DISMISSED** [review]
apparently static finals can be non capitalized

**Review by Wizard603 — APPROVED** [review]
Good catch on the metric names

**Review by craftedswift — APPROVED** [review]

**Review by dozysushi — APPROVED** [review]

**Pull request merged**